internal/provisioner/main.go

```diff
@@ -179,36 +197,17 @@ func main() {
 		os.Exit(1)
 	}
 
-	const plainBundleProvisionerName = "plain"
-	unpacker, err := source.NewDefaultUnpacker(mgr, ns, plainBundleProvisionerName, unpackImage)
+	unpacker, err := source.NewDefaultUnpacker(mgr, ns, provisioner, unpackImage)
 	if err != nil {
 		setupLog.Error(err, "unable to setup bundle unpacker")
 		os.Exit(1)
 	}
 
-	if err = (&controllers.BundleReconciler{
-		Client:     mgr.GetClient(),
-		Scheme:     mgr.GetScheme(),
-		Storage:    bundleStorage,
-		Finalizers: bundleFinalizers,
-		Unpacker:   unpacker,
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", rukpakv1alpha1.BundleKind)
-		os.Exit(1)
-	}
-
-	cfgGetter := helmclient.NewActionConfigGetter(mgr.GetConfig(), mgr.GetRESTMapper(), mgr.GetLogger())
-	if err = (&controllers.BundleDeploymentReconciler{
-		Client:             mgr.GetClient(),
-		Scheme:             mgr.GetScheme(),
-		BundleStorage:      bundleStorage,
-		ReleaseNamespace:   ns,
-		ActionClientGetter: helmclient.NewActionClientGetter(cfgGetter),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", rukpakv1alpha1.BundleDeploymentKind)
+	// Call the relevant provisioner setup function
+	if err := setupProvisioner[provisioner](ns, mgr, bundleStorage, bundleFinalizers, unpacker); err != nil {
```

Comment:
I wonder if, for the sake of clarity, we can attach the reconciler attributes in-line, as before, since it makes it really clear what's happening. This method does save space, but at the expense of being able to see what's happening.